WalletWasabi/Wallets/P2pBlockProvider.cs

```diff
@@ -74,14 +79,35 @@ public async Task<Block> GetBlockAsync(uint256 hash, CancellationToken cancellat
 				cancellationToken.ThrowIfCancellationRequested();
 				try
 				{
-					// Try to get block information from local running Core node first.
-					block = await TryDownloadBlockFromLocalNodeAsync(hash, cancellationToken).ConfigureAwait(false);
+					if (CoreNode?.RpcClient is not null)
+					{
+						block = await TryDownloadBlockFromRpcAsync(hash, cancellationToken).ConfigureAwait(false);
+					}
+					else
+					{
+						using (await ConnectLocalNodeLock.LockAsync(cancellationToken).ConfigureAwait(false))
+						{
+							await TryConnectToLocalNodeAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
New function to separate the connection code from `TryDownloadBlockFromLocalNodeAsync`